### ISSUE TITLE
Font weights appear slightly different between Flutter and Android, sometimes

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#2943]
A good example is Title, or Body 1. It appears that Android is a bit bolder/thicker than Flutter's rendering here.

Android's text size for title's style: `<dimen name="abc_text_size_title_material">20sp</dimen>`

Flutter's text size and weight: `const TextStyle(inherit: false, fontSize:  20.0, fontWeight: FontWeight.w500, color: Colors.black87, height: 28.0 / 20.0, textBaseline: TextBaseline.alphabetic)`

(Variables that have not yet been checked: same version of the font? 

Flutter screenshot:

![flutter](https://cloud.githubusercontent.com/assets/5479/14054716/aeb97c12-f298-11e5-91e7-65bdb374a3de.png)

Android screenshot:

![android](https://cloud.githubusercontent.com/assets/5479/14054722/b457b404-f298-11e5-8019-932dd283557e.png)

Flutter code:

```
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart' show debugPaintSizeEnabled;

void main() {
  // debugPaintSizeEnabled = true;
  runApp(new MaterialApp(title: 'Flutter Demo', routes: <String, WidgetBuilder>{
    '/': (BuildContext context) => new FlutterDemo()
  }));
}

class FlutterDemo extends StatefulWidget {
  @override
  _FlutterDemoState createState() => new _FlutterDemoState();
}

class _FlutterDemoState extends State<FlutterDemo> {
  @override
  Widget build(BuildContext context) {
    final TextTheme textTheme = Theme.of(context).textTheme;
    final styleItems = [
      new Text('Display 3', style: textTheme.display3),
      new Text('Display 2', style: textTheme.display2),
      new Text('Display 1', style: textTheme.display1),
      new Text('Headline', style: textTheme.headline),
      new Text('Title', style: textTheme.title),
      new Text('Subheading', style: textTheme.subhead),
      new Text('Body 2', style: textTheme.body2),
      new Text('Body 1', style: textTheme.body1),
      new Text('Caption', style: textTheme.caption),
      new Text('BUTTON', style: textTheme.button),
    ];
    return new Scaffold(
        appBar: new AppBar(title: new Text('My Application')),
        body: new Padding(
            padding:
                const EdgeInsets.symmetric(horizontal: 8.0, vertical: 16.0),
            child: new Block(children: styleItems)));
  }
}
```

Android code:

```<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:orientation="vertical"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:paddingBottom="@dimen/activity_vertical_margin"
    android:paddingLeft="@dimen/activity_horizontal_margin"
    android:paddingRight="@dimen/activity_horizontal_margin"
    android:paddingTop="@dimen/activity_vertical_margin"
    app:layout_behavior="@string/appbar_scrolling_view_behavior"
    tools:context="com.example.sethladd.myapplication.MainActivity"
    tools:showIn="@layout/activity_main">

```
<TextView
    android:id="@+id/display3"
    android:singleLine="true"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Display3"
    android:text="Display 3" />

<TextView
    android:id="@+id/display2"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Display2"
    android:text="Display 2" />

<TextView
    android:id="@+id/display1"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Display1"
    android:text="Display 1" />

<TextView
    android:id="@+id/headline"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Headline"
    android:text="Headline" />

<TextView
    android:id="@+id/title"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Title"
    android:text="Title" />

<TextView
    android:id="@+id/subhead"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Subhead"
    android:text="Subheader" />

<TextView
    android:id="@+id/body2"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Body2"
    android:text="Body 2" />

<TextView
    android:id="@+id/body1"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Body1"
    android:text="Body 1" />

<TextView
    android:id="@+id/caption"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Caption"
    android:text="Caption" />

<TextView
    android:id="@+id/button"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    style="@style/Base.TextAppearance.AppCompat.Button"
    android:text="Button" />
```

</LinearLayout>

```
```

**Comment by sethladd** [comment]
For more direct analysis:

![screen shot 2016-03-28 at 9 28 44 am](https://cloud.githubusercontent.com/assets/5479/14082902/81ad9f4c-f4c7-11e5-8fb1-85e7741c08bc.png)

**Comment by krisgiesing** [comment]
Seth and I chatted in person, but just to capture some of the discussion here:

- In general, developers shouldn't assume that Flutter text will render text exactly the same way as a platform implementation
- In the specific case of Android, both the platform and Flutter are aiming to fulfill material design. In addition, they both use Skia. So the rendering will be very close, but still not necessarily exactly the same.
- Developers wanting to port applications from native Android or iOS applications probably care most about making sure the layout is close to what they're porting from. Font typeface, font size, and line spacing are all factors in this. In the above case, the typeface appears to be the same, and the font size is mostly consistent (although there are some glyphs that render with somewhat wider stroke widths - the reason for this isn't entirely clear). However the line spacing is obviously different. Seth is looking into capturing the Android defaults so that we know what developers would need to do in Flutter if they wanted to replicate the line spacing of existing designs.
- The "uncanny valley" of looking subtly different from the platform is a risk for Flutter, particularly on iOS where the text rendering engine is very different. This is an ongoing subject of discussion and won't be resolved via any short term fixes.

**Comment by Hixie** [comment]
I wouldn't be surprised here if the difference was because of subpixel alignment differences. I recommend testing this by forcing the position of the text to be exactly the same in both cases.

**Comment by krisgiesing** [comment]
Is this actually a bug?

**Comment by sethladd** [comment]
I don't think it's a bug. It's an "open question that would benefit from additional research and follow up". Not sure how to categorize that, though. I don't want to close it just yet...

**Comment by Hixie** [comment]
I'm going to close this since it's not actionable. If we do the additional research, we should file a new bug with the specifics.

**Comment by kent1ukr** [comment]
any updates on it?
I have an app that uses custom font. And that font looks much more bolder when I use flutter to render it on a screen.
I've even made a "helloWorld" app to see how it looks on Android native, and it looks just fine.
can provide screenshots if needed

**Comment by gouthamravella** [comment]
Any update on this? I don't think this feature is still not available in Flutter 2.0 as well.
